### DISCUSSION TITLE
Add step keyword 'minimum-time', allowing for step failure if it completes too quickly

### DISCUSSION THREAD

**Discussion by MSP-Greg** [community/community#126651] [Actions]
### Select Topic Area

Product Feedback

### Body

Recently a cross-platform action made a change that caused Windows PowerShell to essentially ignore a step's 'run' string without raising an error.  This caused a false positive 'success' for all Windows workflow jobs.

If a 'minimum-time' keyword that failed the step if it completed in less time than the keyword value, such false positives could be avoided.

I'm aware that such false positives are uncommon, but when the 'singular' green check appears, very few people will feel a need to look beyond to a job summary or the logs...

**Comment by grandmas-favorite** [comment]
Hi @MSP-Greg!

We appreciate your feedback and your valuable input. We want to be transparent about our current priorities. You can gain insights into the projects we’re currently focusing on for the upcoming year by checking our [public roadmap](https://github.com/github/roadmap/projects/1). While smaller fixes may not always be explicitly listed, they often get addressed when related work is in progress.

While we do review all incoming feedback unfortunately due to the volume of feedback, we may not be able to respond to each one individually at this time. However, we strongly encourage you to continue providing your valuable input as it greatly helps us prioritize and improve GitHub.

Your contributions are instrumental in making GitHub better for all users!
